tests/suite/upstream_settings_test.go

```diff
@@ -0,0 +1,525 @@
+					func() error {
+						return expectRequestToSucceed(baseCoffeeURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(1000 * time.Millisecond).
+					Should(Succeed())
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseTeaURL, address, "URI: /tea")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(1000 * time.Millisecond).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+
+			BeforeAll(func() {
+				podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+
+				ngfPodName := podNames[0]
+
+				conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("Coffee upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "upstream",
+						Value:     "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "zone",
+						Value:     "uspolicy_coffee_80 512k",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive",
+						Value:     "100",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_requests",
+						Value:     "55",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_time",
+						Value:     "1m",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_timeout",
+						Value:     "5h",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+				}),
+				Entry("Tea upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "zone",
+						Value:     "uspolicy_tea_80 128k",
+						Upstream:  "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "upstream",
+						Value:     "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+				}),
+			)
+		})
+	})
+
+	When("UpstreamSettingsPolicy targets a Service that does not exist", func() {
+		Specify("upstreamSettingsPolicy sets no condition", func() {
+			files := []string{"upstream-settings-policy/invalid-svc-usps.yaml"}
+
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			uspolicyNsName := types.NamespacedName{Name: "usps-target-not-found", Namespace: namespace}
+			ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetStatusTimeout)
+			defer cancel()
+
+			err := wait.PollUntilContextCancel(
```

Comment:
@sjberman @salonichf5 we need Consistently in this case because we are trying to test that over a period of time we don't write status to this policy. Poll does not work here, because we will only check that the policy doesn't have ancestors once, and then we will return successfully.

Reply:
Ah I see, I was confusing Consistently with Eventually.

Reply:
I have updated it to `Consistently` matcher. My bad, I assumed you knew that's what we were trying to do. I should have clarified as well.